### ISSUE TITLE
BeautifulSoup invocation erroneously alters the resulting html

### ISSUE THREAD

**Issue by akhmerov** [danielfrg/pelican-ipynb#56]
[this](https://github.com/danielfrg/pelican-ipynb/blob/master/core.py#L79) bit of code seems to be harmful. 

I have observed it producing from this:
```markdown
line with a line break (so ending with a double space)  
line without a line break
```
the following erroneous html:
```html
line with a line break (so ending with a double space)<br>
line without a line break</br>
```
(note the `</br>` tag). This results in large extra empty space added to a text with a lot of line breaks.

**Comment by ischurov** [comment]
Agree. Moreover: it makes almost impossible to include HTML code inside code blocks as `soup.decode(formatter=None)` replaces *all* entities (including &lt; &rt;) to its corresponding symbols. So if I have something like `print("<b>")` in the source I get actual <b> tag in the output. And this is unavoidable in general, as BeautulSoup converts entities to corresponding Unicode symbols on parsing and therefore losses some information.

This part of code is used to remove all cells with `#ignore` text. Personally, I'm willing just to comment it out as this feature is not crucial for me. Nevertheless, I'm not sure how to solve this problem better.

**Comment by ischurov** [comment]
Most probably it is better not to tweak with HTML tree but to remove the corresponding cells from ipynb JSON file before invoking `HTMLExporter`.

**Comment by akhmerov** [comment]
Seems correct, an nbconvert `Preprocessor` is the way to go. I didn't need the feature, so I just removed it entirely.
